tracing/zipkin/zipkin.go

```diff
@@ -259,6 +312,12 @@ func FromContext(ctx context.Context) (*Span, bool) {
 	return span, true
 }
 
+// CollectorToContext stores a Zipkin collector to be used when annotating
+// resources using their own child spans with NewChildSpan.
+func CollectorToContext(ctx context.Context, c Collector) context.Context {
+	return context.WithValue(ctx, CollectorContextKey, c)
+}
```

Comment:
So, until now, tracing was exclusively managed by endpoint middlewares. The collector was given as a dependency to the middleware during construction, and the span(s) were passed through the context because they needed to cross various boundaries and that's the only way to do it.

But if you need to do tracing stuff within your service methods, then by definition tracing is now part of your business logic, and the collector should be an explicit dependency of your service and fed as part of the constructor. Basically I am objecting to passing anything more than dumb data through the context object. It's a little bit of a gray area with the collector, but I want to hold a really strict line here, so we don't start seeing loggers or database connections passed thru the context too.

Does this make sense? Can NewChildSpan etc. be refactored?